forte/data/data_store.py

```diff
@@ -477,21 +477,40 @@ def _delete_entry_by_loc(self, type_name: str, index_id: int):
             )
         target_list.pop(index_id)
 
-    def get_entry(self, tid: int) -> Tuple[List, int, int]:
+    def get_entry(self, tid: int) -> Tuple[List, str, int]:
         r"""This function finds the entry with ``tid``. It returns the entry,
         its ``type_name``, and the index in the ``type_name`` list.
 
         Args:
             tid (int): Unique id of the entry.
 
         Returns:
-            The entry which ``tid`` corresponds to, its ``type_id`` and its index
-            in the ``type_name`` list.
+            The entry which ``tid`` corresponds to, its ``type_name`` and its
+            index in the ``type_name`` list.
         """
+        if tid not in self.__entry_dict:
+            raise ValueError(f"Entry with tid {tid} not found.")
+        entry = self.__entry_dict[tid]
+        entry_type = entry[constants.ENTRY_TYPE_INDEX]
+        if entry_type not in self.__elements:
+            raise KeyError(f"Entry of type {entry_type} is not found.")
+
         # If the entry is an annotation, bisect the annotation sortedlist
         # to find the entry. May use LRU cache to optimize speed.
         # Otherwise, use ``index_id`` to find the index of the entry.
-        raise NotImplementedError
+        index_id = -1
+        if self._is_annotation(entry_type):
```

Comment:
If we don't need `index_id`, does it mean that we don't need to bisect the list? The `entry` content is already obtained through `self.__entry_dict`.

It seems that we only need `index_id` when we are doing operations like `prev`, `next`, or some sort of iteration. If we simply want to take the entry value, `tid` lookup seems to be sufficient.

Also, @wanglec @qinzzz any comments?

Reply:
Yes, if we only need the `entry list` and `entry name`, we can directly get them from `entry_dict`. We can probably add a separate function for getting `index_id`?

Reply:
Since `delete_entry` function also needs to get `index_id`, we can implement this in a new pull request.

Reply:
we can actually add a function in this PR since (prev, next) will need the id.